packages/expo-random/android/src/main/java/expo/modules/random/RandomModule.java

```diff
@@ -1,39 +1,40 @@
 package expo.modules.random;
 
-import android.content.Context;
 import android.util.Base64;
 
 import java.security.SecureRandom;
+import java.security.NoSuchAlgorithmException;
 
-import org.unimodules.core.ExportedModule;
-import org.unimodules.core.ModuleRegistry;
-import org.unimodules.core.interfaces.ExpoMethod;
-import org.unimodules.core.Promise;
+import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
 
-public class RandomModule extends ExportedModule {
-
-  SecureRandom mRandom;
-
-  public RandomModule(Context context) {
+public class RandomModule extends ReactContextBaseJavaModule {
+  public RandomModule(ReactApplicationContext context) {
     super(context);
   }
 
-  @Override
-  public void onCreate(ModuleRegistry moduleRegistry) {
-    if (moduleRegistry != null) {
-      mRandom = new SecureRandom();
-    }
-  }
-
   @Override
   public String getName() {
     return "ExpoRandom";
   }
 
-  @ExpoMethod
-  public void getRandomBase64StringAsync(int randomByteCount, final Promise promise) {
+  private String _getRandomBase64String(int randomByteCount) throws NoSuchAlgorithmException {
     byte[] output = new byte[randomByteCount];
-    mRandom.nextBytes(output);
-    promise.resolve(Base64.encodeToString(output, Base64.NO_WRAP));
+    SecureRandom random = new SecureRandom();
+    random.nextBytes(output);
+
+    return Base64.encodeToString(output, Base64.NO_WRAP);
+  }
+
+  @ReactMethod
+  public void getRandomBase64StringAsync(int randomByteCount, Promise promise) throws NoSuchAlgorithmException {
```

Comment:
Wouldn't we want to catch the `NoSuchAlgorithmException` and reject the `Promise` with a developer-readable error? I don't remember how this would behave if we just threw an exception. (The same question applies to sync equivalent.)

Reply:
@ide mentioned:

> Should be able to remove the NoSuchAlgorithmException declarations throughout this file since we just use the default RNG algorithm.

so i went ahead and did this